packages/sail_ui/lib/widgets/inputs/menu_items.dart

```diff
@@ -0,0 +1,207 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:sail_ui/sail_ui.dart';
+
+const _menuItemHeight = 24.0;
+const _menuDividerHeight = 10.0;
+const _menuPadding = EdgeInsets.all(4);
+const _menuPaddingWindows = EdgeInsets.symmetric(vertical: 4);
+
+class SailMenu extends StatelessWidget {
+  const SailMenu({
+    required this.items,
+    this.width,
+    super.key,
+  });
+
+  final List<SailMenuEntity> items;
+  final double? width;
+
+  double get height => items.fold(_menuPadding.vertical * 2, (prev, e) => prev + e.height);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      decoration: BoxDecoration(
+        color: SailTheme.of(context).colors.popoverBackground,
+        borderRadius: BorderRadius.all(Radius.circular(context.isWindows ? 3 : 4)),
+        boxShadow: const [
+          BoxShadow(
+            color: Colors.black26,
+            blurRadius: 16,
+            offset: Offset(0, 4),
+          ),
+        ],
+        border: Border.all(
+          color: Colors.grey,
+          width: 0.5,
+        ),
+      ),
+      padding: context.isWindows ? _menuPaddingWindows : _menuPadding,
+      child: IntrinsicWidth(
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: items,
+        ),
+      ),
+    );
+  }
+}
+
+abstract class SailMenuEntity extends Widget {
+  const SailMenuEntity({super.key});
+
+  double get height;
+}
+
+class SailMenuItem extends StatefulWidget implements SailMenuEntity {
+  const SailMenuItem({
+    required this.child,
+    this.onSelected,
+    this.height = _menuItemHeight,
+    super.key,
+  });
+
+  final Widget child;
+  final VoidCallback? onSelected;
+
+  @override
+  final double height;
+
+  @override
+  State<SailMenuItem> createState() => _SailMenuItemState();
+}
+
+class _SailMenuItemState extends State<SailMenuItem> {
+  bool _hover = false;
+  bool _selected = false;
+  bool _flashing = false;
+
+  Timer? _closeTimer;
+  Timer? _toggleFlashTimer;
+
+  @override
+  Widget build(BuildContext context) {
+    bool enabled = widget.onSelected != null;
+    var menuPadding = context.isWindows ? _menuPaddingWindows : _menuPadding;
+
+    var menu = context.findAncestorWidgetOfExactType<SailMenu>();
+    var menuWidth = menu?.width;
+    if (menuWidth != null) menuWidth += menuPadding.horizontal * 2;
+
+    TextStyle textStyle;
+    if (enabled) {
+      textStyle = SailStyleValues.eleven.copyWith(
+        color: (_hover || (_selected && _flashing)) && !context.isWindows ? Colors.white : SailTheme.of(context).colors.text,
+      );
+    } else {
+      textStyle = SailStyleValues.eleven.copyWith(
+        color: SailTheme.of(context).colors.textTertiary,
+      );
+    }
```

Comment:
Same as above. SailStyleValues are not intended to be copied like this. You're only changing the color